x-pack/legacy/plugins/maps/public/layers/sources/es_geo_grid_source/geo_tile_utils.js

```diff
@@ -87,7 +87,13 @@ function sec(value) {
 }
 
 function latitudeToTile(lat, tileCount) {
-  const radians = (lat * Math.PI) / 180;
+  let boundedLat = lat;
```

Comment:
we're doing a similar clamping here to make mapbox happy for the same reasons.

https://github.com/elastic/kibana/blob/59c90f15cb3243b67bd2abdc61070cc40b677acd/x-pack/legacy/plugins/maps/public/connected_components/map/mb/view.js#L236-L243

mind extracting to utility or configure 89 constants (?). Fwiw, 89.9 would fall outside the tile-bounds anyway.